### ISSUE TITLE
Authentication and Authorization Functionality

### ISSUE THREAD

**Issue by soham0005** [SB2318/UltimateHealth#18]
# Implementation of Login and Signup Routes for the backend

```
user:{
    user_id: (Random)
    user_name: 'string',
    user_handle:' string'
    email:'string'
    user_role:'string' [Doctor, General]
    profile_image:'string',
    created_at:Date,
    last_updated_at:'Date
},

Doctor:{
      user_id: (Random)
    user_name: 'string',
    user_handle:' string'
    email:'string'
    user_role:'string' 
    specialization:'string',
    qualification:'string',
    years_of_experience:'string',
    contact_detail:{
      phone_no: number,
      email_id:"string"
     },
    profile_image:'string',
    created_at:Date,
    last_updated_at:'Date
}
```

# PR Submission:

You need to test the server in Postman and attach the snapshot of the response with your PR.

Continuation of  #21 

Please start work, only when #21 will complete.
Thanks for your patience.

**Comment by soham0005** [comment]
Heyy @SB2318 , can i work on the functionality of login and signup routes for the backend.

Solution Provided:
1) Login route with validations
2) Signup Route with validations

**Comment by Aditya6112** [comment]
Can I work on this issue?
@SB2318

**Comment by rushiii3** [comment]
Hey @SB2318, i would like to contribute for making a robust authentication system!

**Comment by SB2318** [comment]
@soham0005 Please wait for a few minutes. I will inform you soon!

**Comment by SB2318** [comment]
@soham0005 You can start work.

**Comment by soham0005** [comment]
@SB2318 Thank you for the concern, I have already started to work on it, as my exams were going on in the past two days, I couldn't update you with the details.

**Comment by princyym** [comment]
Hello @SB2318 Can you assign me this issue, I want to work on this issue.

**Comment by SB2318** [comment]
@soham0005 do you need any in-voice discussion, let me know in the project channel, And please try to give me an update

**Comment by soham0005** [comment]
@SB2318 sure, we'll connect there, as for the updates, the backend structure was not quite well structured, I have made several changes which we can discuss, also right now I am working on JWT for login

**Comment by SB2318** [comment]
I am available now, please message

**Comment by soham0005** [comment]
check once if the message is recieved or not

**Comment by SB2318** [comment]
> @soham0005 You can start work. I believe in your ability, but still If you find any stuck situation, feel free to connect with me.

@soham0005  This issue was on hold, so your counting will start from when I informed you to start your work.

**Comment by SB2318** [comment]
> Hello @SB2318 Can you assign me this issue, I want to work on this issue.

No worries, new issues will come. 😇

**Comment by SB2318** [comment]
> Can I work on this issue? @SB2318

@Aditya6112 No worries, new issues will come. 😇

**Comment by SB2318** [comment]
closed by #53
